.github/workflows/test.yml

```diff
@@ -0,0 +1,88 @@
+name: KBase Catalog test
+
+on:
+  pull_request:
+    types:
+    - opened
+    - reopened
+    - synchronize
+    - ready_for_review
+  push:
+    # run workflow when merging to main or develop
+    branches:
+      - main
+      - master
+      - develop
+
+jobs:
+  catalog_tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - python-version: '3.7'
+            mongo-version: '3.6'
+          # - python-version: '3.7'
+          #   mongo-version: '7.0.4'
+    services:
+      mongo:
+        image: mongo:${{matrix.mongo-version}}
+        ports:
+            - 27017:27017
+        options: --name mongo${{matrix.mongo-version}}
+
+    steps:
+      - name: Repo checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{matrix.python-version}}
+
+      - name: Install dependencies and set up test config
+        shell: bash
+        env:
+          KBASE_CI_TOKEN: ${{ secrets.KBASE_CI_TOKEN }}
+          ADMIN_USER: ${{ secrets.KBASE_BOT_USER_CI }}
+
+        run: |
```

Comment:
Looking at travis.yml, there's ant and jars install steps which I assume are for NMS, which is required for tests. How do the tests pass now without this step?

Reply:
submodules got pulled recursively in the repo checkout step, including jars repo. No need to install jars step repetitively.

https://github.com/kbase/catalog/actions/runs/12362708941/job/34502579817?pr=136#step:3:99

Reply:
:+1: